xdem/spatial_tools.py

```diff
@@ -11,6 +11,55 @@
 import rasterio.warp
 
 
+def get_mask(array: Union[np.ndarray, np.ma.masked_array]) -> np.ndarray:
+    """
+    Return the mask of invalid values, whether array is a ndarray with NaNs or a np.ma.masked_array.
+
+    :param array: Input array.
+
+    :returns invalid_mask: boolean array, True where array is masked or Nan.
+    """
+    return (array.mask | np.isnan(array.data)) if isinstance(array, np.ma.masked_array) else np.isnan(array)
```

Comment:
Funny, I had just made a similar function in coreg.py! I suggest that this is named `_get_mask()` instead to hide it from the public API.

Reply:
we should replace `np.isnan` intp `np.isfinite` here or we might run into errors in some cases (it's rare but happens, I had to do this in pybob a couple years ago)

Reply:
Added in c4c7e20